### PR TITLE
[Impeller] Specify the min simulator version.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Fixes https://github.com/flutter/flutter/issues/113358

### CODE REVIEW

File: impeller/tools/build_metal_library.py [hunk 1]
```diff
@@ -100,15 +100,19 @@ def main():
...
+        # see https://github.com/flutter/flutter/issues/106066
+        '--std=ios-metal1.2',
     ]
     if args.platform == 'ios':
-      # limiting to ios-metal1.2 disables shader debug symbols, only
-      # enabling these in optimize mode.
-      # see https://github.com/flutter/flutter/issues/106066
       command += [
-          '--std=ios-metal1.2',
           '-mios-version-min=10.0',
```
Comment (zanderso) [inline impeller/tools/build_metal_library.py:110]: Should this also be 11.0?

**Comment by chinmaygarde** [comment]
@zanderso I've tried to test this on hosts I have on hand but they were working before too. I am unable to remote into my desktop in the office that I am pretty sure will tickle the error. If you are able to test this fixes the error, I'd appreciate a sanity check. This builds and works just fine.

File: impeller/tools/build_metal_library.py (hunk 1 above)
Comment (chinmaygarde) [reply impeller/tools/build_metal_library.py:110]: Yeah, may I change it in a later patch though? It is unrelated to the sim fix.

Comment (chinmaygarde) [reply impeller/tools/build_metal_library.py:110]: I also want to remove the workaround around optimizations that were making Xcode crashy. This script is looking a bit confusing now.

**Review by zanderso — APPROVED** [review]
This fixes the issue for me locally, and the validation errors now no longer show up in the scenario test logs https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8800025623849152401/+/u/test:_Scenario_App_Integration_Tests/stdout?format=raw

**Pull request merged**